File: worktree-pipeline/src/main/java/com/smartwallet/alert/repository/AlertRepository.java
package com.smartwallet.alert.repository;

import com.smartwallet.alert.entity.Alert;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface AlertRepository extends JpaRepository<Alert, Long> {
    List<Alert> findByUserIdAndIsActiveTrue(Long userId);
    List<Alert> findByUserIdAndAssetSymbol(Long userId, String assetSymbol);
    List<Alert> findByAssetSymbolAndIsActiveTrue(String assetSymbol);
}

File: src/main/kotlin/com/smartwallet/watchlist/repository/AlertRepository.kt
package com.smartwallet.watchlist.repository

import com.smartwallet.watchlist.model.Alert
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface AlertRepository : JpaRepository<Alert, Long> {
    fun findAllByUserId(userId: Long): List<Alert>
}
